core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java

```diff
@@ -299,55 +343,125 @@ private void throwIfFailed()
                 throw failure;
             }
         }
+
+        public int getStagesCount()
+        {
+            return descriptors.rowMap().size();
+        }
+
+        public Stream<Long> getStagesReservedBytes()
+        {
+            return stagesReservedBytes.values().stream()
+                    .map(AtomicLong::get);
+        }
     }
 
-    private static class TaskDescriptorKey
+    public static class StorageStats
```

Comment:
Curious, is it possible to expose a "managed record"?

Reply:
lets skip to until never